### PR TITLE
Modified the source for wider range of inputs

### PR DESCRIPTION
**Pull request by Gpamangkp**
#### Description of Change

Modified the `graph/breadth_first_search.cpp` file to include wider range or inputs by creating a generic class

1. Replaced the namespace with a generic class Graph
2. Created add edge function to account for both directional and bidirectional graph
3. test case to include string nodes also

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1359"></a>

### CODE REVIEW

**Comment by Gpamangkp** [comment]
@Panquesito7 please review this and suggest any changes if applicable

**Review by Panquesito7 — COMMENTED** [review]
Currently, the previous code is very well documented.
Would you be able to provide documentation here?

Also, do not remove namespaces. They help in the Doxygen documentation. 🙂

File: graph/breadth_first_search.cpp [hunk 1]
```diff
@@ -5,6 +5,8 @@
  * (Breadth First Search)](https://en.wikipedia.org/wiki/Breadth-first_search)
  *
  * \author [Ayaan Khan](http://github.com/ayaankhan98)
+ * \contributor [Aman Kumar Pandey](http://github.com/gpamangkp)
```
Comment (Panquesito7) [inline graph/breadth_first_search.cpp:8]:
```suggestion
 * \author [Aman Kumar Pandey](https://github.com/gpamangkp)
```

File: graph/breadth_first_search.cpp [hunk 2]
```diff
@@ -5,6 +5,8 @@
  * (Breadth First Search)](https://en.wikipedia.org/wiki/Breadth-first_search)
  *
  * \author [Ayaan Khan](http://github.com/ayaankhan98)
```
Comment (Panquesito7) [inline graph/breadth_first_search.cpp:7]:
```suggestion
 * \author [Ayaan Khan](https://github.com/ayaankhan98)
```

File: graph/breadth_first_search.cpp [hunk 3]
```diff
@@ -48,150 +50,151 @@
...
-            /// if the vertex is not visited then mark it as visited
-            /// and push it to the queue
-            if (!visited[x]) {
-                visited[x] = true;
-                tracker.push(x);
+#include <map>
+#include <list>
+#include <string>
+
+///namespace graph which contains the class Graph
```
Comment (Panquesito7) [inline graph/breadth_first_search.cpp:57]:
```suggestion
/**
 * \namespace graph
 * \brief Graph algorithms
 */
```

File: graph/breadth_first_search.cpp [hunk 4]
```diff
@@ -48,150 +50,151 @@
...
+          }
         }
-    }
-    return visited;
+        return visited;
+      }
+  };
+  /* Class definition ends */
 }
+///Namespace definition over
```
Comment (Panquesito7) [inline graph/breadth_first_search.cpp:131]:
```suggestion
```

File: graph/breadth_first_search.cpp [hunk 5]
```diff
@@ -48,150 +50,151 @@
...
             }
+          }
         }
-    }
-    return visited;
+        return visited;
+      }
+  };
+  /* Class definition ends */
 }
```
Comment (Panquesito7) [inline graph/breadth_first_search.cpp:130]:
```suggestion
}  // namespace graph
```

**Comment by Gpamangkp** [comment]
@Panquesito7 I have modified as per your suggestions.

**Review by ayaankhan98 — APPROVED** [review]
LGTM! @Gpamangkp Thanks for your contribution

**Comment by Panquesito7** [comment]
@ayaankhan98 please use squash-merge. Thanks. 🙂

**Pull request merged**